lib/services/elasticsearch.js

```diff
@@ -264,6 +264,47 @@ module.exports = function (kuzzle, options) {
   };
 
   /**
+   * Replace a document to ElasticSearch
+   *
+   * @param {RequestObject} requestObject
+   * @returns {Promise} resolve an object that contains _id
+   */
+  this.replace = function (requestObject) {
+    var
+      deferred = q.defer(),
+      data = cleanData.call(this, requestObject);
+
+    this.client.exists({
+      index: data.index,
+      type: data.type,
+      id: data.id
+    })
+      .then((exists) => {
+        if (exists) {
+          this.client.index(data)
+            .then(function (result) {
+              indexCache.add.call(kuzzle, data.index, data.type);
```

Comment:
Since the document exists, then so do its index and collection. So there is no need to add them in the cache.

Moreover, the index cache has been overhauled in PR #137